### ISSUE TITLE
FR: Display lyrics for selected file when playback is stopped, but for playing track when playing

### ISSUE THREAD

**Issue by redactedscribe** [jacquesh/foo_openlyrics#310]
Similar to Waveform Minibar (mod):

![foobar2000_2023-10-25_21-18-58](https://github.com/jacquesh/foo_openlyrics/assets/25013746/38b9b952-1563-4841-9f0a-88641c47c1b0)

This would mean that when there is no playback, the component will display the lyrics for whichever file is selected (and thus its background image etc). However, when a track is playing, only show the lyrics, background, etc of the playing track. I find this to be the most useful of the options which Waveform Minibar (mod) provides, but you could also implement the some of the other modes, such as "Selected track". "Playing track, blank when stopped" is currently how foo_openlyrics operates.

Thanks.

**Comment by jacquesh** [comment]
Yeah we could probably do this. It'd basically be an extension of the existing "Show lyrics" button in the playlist, which only searches local sources.

That said, it'd probably be a non-trivial amount of work because the assumption that displayed lyrics are for the currently-playing track is fairly deeply baked in.

**Comment by redactedscribe** [comment]
> Yeah we could probably do this. It'd basically be an extension of the existing "Show lyrics" button in the playlist, which only searches local sources.

I'm not sure how that relates as I don't use that. I do only use local lyrics however.

> That said, it'd probably be a non-trivial amount of work because the assumption that displayed lyrics are for the currently-playing track is fairly deeply baked in.

I see. Well, perhaps it's come to fruition one day. Currently the lyrics window for me is kind of dead space unless something's being played. Even if the lyrics themselves weren't showing, seeing the background for what is clicked / being played would make it more appealing -- maybe this doesn't have the same problem as the lyrics?

**Comment by jacquesh** [comment]
> Even if the lyrics themselves weren't showing, seeing the background for what is clicked / being played would make it more appealing -- maybe this doesn't have the same problem as the lyrics?

Yes this is probably simpler as we don't (currently) do any multithreaded processing for album art, we just ask fb2k for the image and it calls us back later with the data. Tbh I'd probably do these together anyway though, just because the retrieval code for the two are closely related.
